db/db_impl_write.cc

```diff
@@ -62,6 +62,11 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
                          WriteBatch* my_batch, WriteCallback* callback,
                          uint64_t* log_used, uint64_t log_ref,
                          bool disable_memtable) {
+  // The current implementation does not support sync with concurrenet writes
+  assert(!concurrent_writes_ || !write_options.sync);
```

Comment:
We don't tend to assert user requests, unless they are used in radically wrong way, like calling key() with !Valid(). The risk is that we crash users' whole service with a per request misconfig if they run in debug mode.